src/main/java/com/scalar/database/storage/cassandra4driver/SelectStatementHandler.java

```diff
@@ -0,0 +1,250 @@
+import com.datastax.oss.driver.api.querybuilder.relation.Relation;
+import com.datastax.oss.driver.api.querybuilder.select.Select;
+import com.datastax.oss.driver.api.querybuilder.select.SelectFrom;
+import com.scalar.database.api.Get;
+import com.scalar.database.api.Operation;
+import com.scalar.database.api.Scan;
+import com.scalar.database.api.Selection;
+import com.scalar.database.io.Key;
+import com.scalar.database.io.Value;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A handler class for select statement
+ *
+ * @author Hiroyuki Yamada, Yuji Ito
+ */
+@ThreadSafe
+public class SelectStatementHandler extends StatementHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SelectStatementHandler.class);
+
+  /**
+   * Constructs {@code SelectStatementHandler} with the specified {@code CqlSession}
+   *
+   * @param session session to be used with this statement
+   */
+  public SelectStatementHandler(CqlSession session) {
+    super(session);
+  }
+
+  @Override
+  @Nonnull
+  protected PreparedStatement prepare(Operation operation) {
+    checkArgument(operation, Get.class, Scan.class);
+    Select select = null;
+
+    if (operation instanceof Get) {
+      select = prepare((Get) operation);
+    } else {
+      select = prepare((Scan) operation);
+    }
+
+    return prepare(select.asCql());
+  }
+
+  @Override
+  @Nonnull
+  protected BoundStatementBuilder bind(PreparedStatement prepared, Operation operation) {
+    checkArgument(operation, Get.class, Scan.class);
+    BoundStatementBuilder builder = prepared.boundStatementBuilder();
+
+    if (operation instanceof Get) {
+      bind(builder, (Get) operation);
+    } else {
+      bind(builder, (Scan) operation);
+    }
+
+    return builder;
+  }
+
+  @Override
+  @Nonnull
+  protected ResultSet execute(BoundStatement bound, Operation operation) {
+    return session.execute(bound);
+  }
+
+  @Override
+  protected void overwriteConsistency(BoundStatementBuilder builder, Operation operation) {
+    // nothing to overwrite
+  }
+
+  private Select prepare(Get get) {
+    Select select = getSelect(get);
+    return setPredicates(select, get);
+  }
+
+  private Select prepare(Scan scan) {
+    Select select = getSelect(scan);
+    select = createStatement(select, scan);
+
+    Map<String, ClusteringOrder> orderings = getOrderings(scan.getOrderings());
+    if (!orderings.isEmpty()) {
+      select = select.orderBy(orderings);
+    }
+
+    if (scan.getLimit() > 0) {
+      select = select.limit(scan.getLimit());
+    }
+
+    return select;
+  }
+
+  private Select getSelect(Selection sel) {
+    SelectFrom selectFrom = selectFrom(sel.forNamespace().get(), sel.forTable().get());
+
```

Comment:
It's hard to judge but removing the empty line is better and more consistent in this case like `setPredicates` since there are only 2 lines of code ?

Reply:
Fixed